### DISCUSSION TITLE
Update the existing custom metrics

### DISCUSSION THREAD

**Discussion by atifsayeedi** [prometheus/client_python#699] [Q&A]
I have a metrics like this device{deviceID=“dev1”,code=“3”,status=“Connected”,ipAddress="192.180.0.0"} 1.0 and the value of the labels like code and status changes after certain conditions then promethius inserts a new metrics for this.

Please suggest how to update the existing metrics.

device{deviceID=“dev1”,code=“3”,status=“Connected”,ipAddress="192.180.0.0"} 1.0
to
device{deviceID=“dev1”,code=“4”,status=“Not Connected”,ipAddress="192.180.0.0"} 1.0

**Comment by csmarchbanks** [comment]
Hello, the typical way of doing this is by using the idea of a state set, which means you will always export both of the metrics you provided, but only one of them would be 1, and the other one would be 0. Then, when status changes you set the value of the old status to 0, and the new status to 1. The reason for this is to avoid creating very short lived series if a device is alternating between connected and not connected often.

At query time, you can then use > 0 to filter to just the state at any point in time.

**Reply by atifsayeedi** [reply]
Hi Chris, I understand your point, but for some of the value of labels may change with unknown values like ipAddress/macAddress of the device. If ipAddress is changed then promethius will create a new metrics for this but I just wanted to update the existing one or delete this particular metrics and insert new metrics with new ipAddress.

Please suggest how to handle these.

**Reply by csmarchbanks** [reply]
In the case that the IP address is changed you would likely want to delete all of the metrics for the old IP address and then create new metrics with the new IP address.

You could also use a [custom collector](https://github.com/prometheus/client_python/#custom-collectors) to generate the list of metrics at scrape time from an updated list of devices or something like that.

Would one of those solutions work for you?

**Reply by atifsayeedi** [reply]
Hi Chris, could you please suggest how to delete the specific metrics with the old IP address which is changed. So, that I can create new metrics for this with new IP address.

**Reply by csmarchbanks** [reply]
Yep, you can use `metric.remove(['foo', 'bar'])` to remove the series specified by the label values. You can also use `metric.clear()` to remove all the series on a metric.
